### PR TITLE
Fix/ask ai

### PR DESCRIPTION
**Pull request by RybakovaIE**

### CODE REVIEW

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include "sc-agents-common/utils/IteratorUtils.hpp"
 #include "sc-agents-common/utils/GenerationUtils.hpp"
 #include "sc-agents-common/utils/AgentUtils.hpp"
+#include "sc-memory/utils/sc_log.hpp"
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:11]: why do you need it?

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 2]
```diff
@@ -31,9 +32,12 @@ SC_AGENT_IMPLEMENTATION(GenerateResponseAgent)
       utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, otherAddr, scAgentsCommon::CoreKeynodes::rrel_1);
   ScAddr const & answerAddr =
       utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, otherAddr, scAgentsCommon::CoreKeynodes::rrel_2);
+  ScAddr const & conceptAddr =
+      utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, messageAddr, Keynodes::rrel_entity);
 
   ScAddr messageAnswer;
   ScAddr const actionToApply = createActionNode(messageAddr);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionToApply, conceptAddr);
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:40]: It is not correct for all actions to apply. You need some common logic to attach arguments from kb <message, action>. As it seems to me, this logic is in `createActionNode` method

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 3]
```diff
@@ -69,8 +75,9 @@ void GenerateResponseAgent::attachAnswer(
     ScAddr const & messageAddr,
     ScAddr const & answerAddr)
 {
-  utils::GenerationUtils::generateRelationBetween(&m_memoryCtx, messageAddr, messageAnswer, Keynodes::nrel_reply_structure);
-  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosTemp, answerAddr, messageAnswer);
+  utils::GenerationUtils::generateRelationBetween(
+      &m_memoryCtx, messageAddr, messageAnswer, Keynodes::nrel_reply_structure);
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:79]: fix documentation please, this keynode is not in structure
image

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 4]
```diff
@@ -148,4 +152,4 @@ void GenerateResponseAgent::processParamsFromMessage(
 
         mappedRelations.insert(targetRole);
       });
-}
+}
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:155]: add newline. Check everywhere

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 5]
```diff
@@ -69,8 +75,9 @@ void GenerateResponseAgent::attachAnswer(
     ScAddr const & messageAddr,
     ScAddr const & answerAddr)
 {
-  utils::GenerationUtils::generateRelationBetween(&m_memoryCtx, messageAddr, messageAnswer, Keynodes::nrel_reply_structure);
-  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosTemp, answerAddr, messageAnswer);
+  utils::GenerationUtils::generateRelationBetween(
+      &m_memoryCtx, messageAddr, messageAnswer, Keynodes::nrel_reply_structure);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, answerAddr, messageAnswer);
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:80]: Why do you need it?

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp (hunk 5 above)
Comment (RybakovaIE) [reply platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:80]: answerAddr is structuresSet in structure translation agent, which must contain a structure (messageAnswer) for translation agent to find and translate

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 6]
```diff
@@ -69,19 +72,21 @@ void GenerateResponseAgent::attachAnswer(
...
 {
   ScAddr const actionClass = FindResponseActionClass(message);
   ScAddr const actionNode = m_memoryCtx.CreateNode(ScType::NodeConst);
   m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionClass, actionNode);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionNode, conceptAddr);
 
   ScAddrSet mappedRelations;
 
-  processParamsFromMessage(message, actionClass, actionNode, mappedRelations);
+  processParamsFromMessage(message, actionClass, actionNode, mappedRelations);answerAddr
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:89]:
```suggestion
  processParamsFromMessage(message, actionClass, actionNode, mappedRelations);
```
?

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 7]
```diff
@@ -69,19 +72,21 @@ void GenerateResponseAgent::attachAnswer(
...
 {
-  utils::GenerationUtils::generateRelationBetween(&m_memoryCtx, messageAddr, messageAnswer, Keynodes::nrel_reply_structure);
-  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosTemp, answerAddr, messageAnswer);
+  utils::GenerationUtils::generateRelationBetween(
+      &m_memoryCtx, messageAddr, messageAnswer, Keynodes::nrel_reply_structure);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, answerAddr, messageAnswer);
 }
 
-ScAddr GenerateResponseAgent::createActionNode(ScAddr const & message)
+ScAddr GenerateResponseAgent::createActionNode(ScAddr const & message, ScAddr const & conceptAddr)
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:80]: you don't need `conceptAddr` here in parameter, find it in `processParamsFromMessage`

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 8]
```diff
@@ -69,19 +72,21 @@ void GenerateResponseAgent::attachAnswer(
...
 {
   ScAddr const actionClass = FindResponseActionClass(message);
   ScAddr const actionNode = m_memoryCtx.CreateNode(ScType::NodeConst);
   m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionClass, actionNode);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionNode, conceptAddr);
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:85]: same

File: platform-dependent-components/problem-solver/py/modules/clean_text_generation_module/clean_text_generation_agent.py [hunk 9]
```diff
@@ -107,12 +104,17 @@ def run(self, action_element: ScAddr) -> ScResult:
...
         response = client.chat.completions.create(
             model='gpt-3.5-turbo',
             messages=messages,
             temperature=0,
         )
         self.logger.info(f'Successfully cleaned text for you\n: {response}')
         self.logger.info(response.choices[0].message.content)
-        return response.choices[0].message.content      
+        return response.choices[0].message.content   
+
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/py/modules/clean_text_generation_module/clean_text_generation_agent.py:120]: empty line

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

File: platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp [hunk 10]
```diff
@@ -132,6 +130,9 @@ void GenerateResponseAgent::processParamsFromMessage(
 
   SC_LOG_DEBUG(m_memoryCtx.HelperGetSystemIdtf(action));
 
+  ScAddr const & entityAddr = utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, message, Keynodes::rrel_entity);
+  m_memoryCtx.CreateEdge(ScType::EdgeAccessConstPosPerm, actionNode, entityAddr);
```
Comment (MksmOrlov) [inline platform-dependent-components/problem-solver/cxx/generate-response-module/agent/GenerateResponseAgent.cpp:134]: it should be found and generated from kb

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
Fix PR  https://github.com/ostis-ai/ostis-ps-lib/pull/5/files first

**Pull request merged**